### PR TITLE
editoast: move many infra elements to `editoast_schemas`

### PR DESCRIPTION
**Pull request by hamz2a**
Part of #7048

### CODE REVIEW

File: editoast/editoast_common/src/direction.rs
Comment (woshilapin) [inline editoast/editoast_common/src/direction.rs:1]: Why is `Direction` not part of `editoast_schemas`? It's a type directly used in `DirectionalTrackRange`, and therefore part of the RailJSON API if I understand correctly?

**Review by woshilapin — COMMENTED** [review]
This PR is very big and could have been split, at least in commits. All `primitives` could have been done independently, then the infra itself could have been split in multiple chunks. It makes it very hard to review and to verify nothing has been left out. If it even happens that we realize something is wrong starting with this commit (it's always possible tests are not catching something), it's going to be hard digging in that commit to find the culprit. Even for you, it would be much easier to follow-up and be sure nothing is forgotten.

File: editoast/editoast_common/src/direction.rs
Comment (hamz2a) [reply editoast/editoast_common/src/direction.rs:1]: I will fix it.

Comment (hamz2a) [reply editoast/editoast_common/src/direction.rs:1]: Done.

**Comment by hamz2a** [comment]
I will make the changes in multiple chunks.

**Pull request closed without merging**